Order List Management Step 1: Basic Order List Support + MVVM

AmandaRiu opened the pull request:
Partially fixes #1330

## Included in this PR

* Orders List code updated to use `MVVM`
* Orders are now fetched and managed using the new order list management logic
* New loading skeleton implementation
* Load more
* Pull to refresh
* Search and Filter
* Click to open an order
* Updating an order status will cause the order list to refresh

#### Order List Loading

![orders-list-loading](https://user-images.githubusercontent.com/5810477/67451168-841ee580-f5d4-11e9-9e7b-f3f6a2ecc637.gif)

#### Order List Filtering

<img src="https://user-images.githubusercontent.com/5810477/67450341-ede9c000-f5d1-11e9-8e75-9e0ed2bd2e19.gif" width="300"/>

#### Order list Searching 

<img src="https://user-images.githubusercontent.com/5810477/67450351-f510ce00-f5d1-11e9-99e0-eb68c7897af8.gif" width="300"/>

## Not Included

The following items will be handled in a future PR:
* Removing future dated orders (needs to be done in FluxC)
* Offline handling
* Error, loading & empty states
* Unit and instrumentation test updates

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

anitaa1990 commented on WooCommerce/src/main/kotlin/com/woocommerce/android/viewmodel/ScopedAndroidViewModel.kt:
```diff
@@ -0,0 +1,17 @@
+abstract class ScopedAndroidViewModel(
+    defaultDispatcher: CoroutineDispatcher,
+    val appContext: Application
+) : ScopedViewModel(defaultDispatcher)
```
This doesn't seem to be used anywhere at the moment?

AmandaRiu replied:
Ah yes, a holdover from my old implementation. Removed in 08937357b9a41f9fea32a400fbf7454712bd5cfe

anitaa1990 commented on WooCommerce/src/main/kotlin/com/woocommerce/android/util/ThrottleLiveData.kt:
```diff
@@ -0,0 +1,47 @@
+ *
+ * @param <T> The type of data hold by this instance
+ */
+class ThrottleLiveData<T> constructor(
```
TIL 😄

AmandaRiu commented:
@anitaa1990 Thank you for the review! I've fixed the crash in c8e5d784d4405eaced64314258952828fc5edc64, and addressed the code comments as well. Ready for another round.
